### PR TITLE
Adds support for ASM

### PR DESCRIPTION
**Pull request by b00kdev**
Adds support for the ASM object which allows you to specify how to handle unsubscribes.

https://docs.sendgrid.com/api-reference/mail-send/mail-send

### CODE REVIEW

File: src/v3.rs [hunk 1]
```diff
@@ -538,13 +555,33 @@ impl Attachment {
     }
 }
 
+impl ASM {
+    /// Construct a new attachment for this message.
```
Comment (gsquire) [inline src/v3.rs:559]: Update this comment?

File: src/v3.rs [hunk 2]
```diff
@@ -538,13 +555,33 @@ impl Attachment {
...
+        Default::default()
+    }
+
+    /// The unsubscribe group to associate with this email.
+    pub fn set_group_id(mut self, group_id: u32) -> Self {
+        self.group_id = group_id;
+        self
+    }
+
+    /// A set containing the unsubscribe groups that you would like to be displayed on the unsubscribe preferences page.
```
Comment (gsquire) [inline src/v3.rs:570]: Let's add a check here to see if there are greater than 25 items. This appears to be the max per the documentation.

**Review by gsquire — CHANGES_REQUESTED** [review]

File: src/v3.rs (hunk 2 above)
Comment (b00kdev) [reply src/v3.rs:570]: Do you want it to truncate to 25 items or should it return a `Result` so we can handle the error case?

Comment (b00kdev) [reply src/v3.rs:570]: I went with using a `SendgridResult` and added a new entry to the `SendgridError` enum.

**Review by gsquire — APPROVED** [review]

**Comment by gsquire** [comment]
Thanks!

**Comment by gsquire** [comment]
This will be released as part of version 0.19.0.

**Pull request merged**